pkg/k8shandler/visualization.go

```diff
@@ -79,28 +79,35 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateVisualization(proxyCo
 			return
 		}
 
-		kibanaStatus, err := clusterRequest.getKibanaStatus()
-		cluster := clusterRequest.cluster
+		clusterRequest.SetKibanaStatus()
+	}
 
-		if err != nil {
-			return fmt.Errorf("Failed to get Kibana status for %q: %v", cluster.Name, err)
-		}
+	return nil
+}
 
-		printUpdateMessage := true
-		retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			if !compareKibanaStatus(kibanaStatus, cluster.Status.Visualization.KibanaStatus) {
-				if printUpdateMessage {
-					logrus.Infof("Updating status of Kibana")
-					printUpdateMessage = false
-				}
-				cluster.Status.Visualization.KibanaStatus = kibanaStatus
-				return clusterRequest.UpdateStatus(cluster)
+func (clusterRequest *ClusterLoggingRequest) SetKibanaStatus() (err error) {
```

Comment:
the name "UpdateKibanaStatus" would be better given you are not providing any args